__tests__/app.test.js

```diff
@@ -317,6 +317,77 @@ describe("GET /api/users", () => {
     })
   })
 
+  
+  describe("GET /api/reviews (queries)", () => {
+    test("200: Able to sort a query depending on category", () => {
+      return request(app)
+      .get("/api/reviews?category=social deduction")
+      .expect(200)
+      .then(({body}) => {
+        expect(body.reviews).toHaveLength(11)
+        body.reviews.forEach((review) => {
+          expect(review).toMatchObject({
+            review_id: expect.any(Number),
+            title: expect.any(String),
+            category: "social deduction",
+            designer: expect.any(String),
+            owner: expect.any(String),
+            review_img_url: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number)
+        })
+      })
+      })
+    })
+    test("200: Able to sort by a query depending on the values specified", () => {
+      return request(app)
+      .get("/api/reviews?category=social deduction&sort_by=votes&order=ASC")
+      .expect(200)
+      .then(({body}) => {
+        expect(body.reviews).toHaveLength(11)
+        body.reviews.forEach((review) => {
+          expect(review).toMatchObject({
+            review_id: expect.any(Number),
+            title: expect.any(String),
+            category: "social deduction",
+            designer: expect.any(String),
+            owner: expect.any(String),
+            review_img_url: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number)
+        })
+        })
+        expect(body.reviews).toBeSortedBy("votes", {
+          descending: false
+        })
+      })
+    })
+    test("400: Invalid sort query if an improper sort by query is fed", () => {
+      return request(app)
+      .get("/api/reviews?category=social deduction&sort_by=wronganswer")
+      .expect(400)
+      .then(({body}) => {
+        expect(body.msg).toBe("Invalid sort query")
+      })
+    })
+    test("400: Invalid order query if improper order query is fed", () => {
+      return request(app)
+      .get("/api/reviews?category=social deduction&sort_by=votes&order=highest")
+      .expect(400)
+      .then(({body}) => {
+        expect(body.msg).toBe("Invalid order query")
+      })
+    })
+    test("404: If a valid query is fed but there is no results found", () => {
+      return request(app)
+      .get("/api/reviews?category=valid request")
+      .expect(404)
+      .then(({body}) => {
+        expect(body.msg).toBe("Category not found")
+      })
+    })
+  })
+
```

Comment:
There is one more situation to consider, a valid and existent category query that just so happens to have no reviews associated with it yet. This isn't an error as nothing has gone wrong so we need to separate that logic out from any error handling that tests for an empty response. We may need to check if the category exists.